### PR TITLE
feat: add embed builder and migrate current commands to embeds

### LINKED ISSUE #38: Update embeds
Currently, we just use texts instead of embeds this should change.
Copy the old embed/emotes utils from bankobot and change that
@warriorzz and @NyCodeGHG should approve that as I copied it from old votebot in bankobot

### PR DESCRIPTION
**Pull request by Niggelgame**
resolves #38

### CODE REVIEW

**Comment by DRSchlaubi** [comment]
Run `gradlew ktlintFormat` to fix the CI

File: src/main/kotlin/dev/schlaubi/votebot/commands/ClaimPermissionsCommand.kt [hunk 1]
```diff
@@ -35,7 +36,7 @@ object ClaimPermissionsCommand : SingleCommand() {
 
     override suspend fun execute(context: Context) {
         if (Permission.Administrator !in context.executor.asMember().getPermissions()) {
-            context.respond("You need the `ADMINISTRATOR` permission to execute this command")
+            context.respond(Embeds.error("You need the `ADMINISTRATOR` permission to execute this command", description = null))
```
Comment (DRSchlaubi) [inline src/main/kotlin/dev/schlaubi/votebot/commands/ClaimPermissionsCommand.kt]: The description is null per default

File: src/main/kotlin/dev/schlaubi/votebot/core/VoteBotImpl.kt [hunk 2]
```diff
@@ -19,6 +19,7 @@
 
 package dev.schlaubi.votebot.core
 
+import dev.kord.common.entity.PresenceStatus
```
Comment (DRSchlaubi) [inline src/main/kotlin/dev/schlaubi/votebot/core/VoteBotImpl.kt]: Why was this added?

File: src/main/kotlin/dev/schlaubi/votebot/util/Colors.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+import dev.kord.common.Color
+
+/**
+ * Wrapper for [Discordapp.com/branding][https://discordapp.com/branding] colors and some other colors:
+ */
+@Suppress("KDocMissingDocumentation", "unused", "MagicNumber")
+object Colors {
+    // Discord
+    val BLURLPLE: Color = Color(114, 137, 218)
```
Comment (DRSchlaubi) [inline src/main/kotlin/dev/schlaubi/votebot/util/Colors.kt]: Those are actually wrong update them to match the new branding

File: src/main/kotlin/dev/schlaubi/votebot/util/Embeds.kt [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+@Suppress("unused", "TooManyFunctions")
+object Embeds {
+
+    /**
+     * Creates a info embed with the given [title] and [description] and applies the [builder] to it.
+     * @see EmbedCreator
+     * @see EmbedBuilder
+     */
+    fun info(title: String, description: String? = null, builder: EmbedCreator = {}): EmbedBuilder =
+        EmbedBuilder().apply {
```
Comment (DRSchlaubi) [inline src/main/kotlin/dev/schlaubi/votebot/util/Embeds.kt]: use the global scope embed function here instead

File: src/main/kotlin/dev/schlaubi/votebot/util/Embeds.kt [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+//    private fun formatSubCommandUsage(command: AbstractSubCommand): String {
+//        val builder = StringBuilder(Constants.firstPrefix)
+//        builder.append(' ').append(command.name).append(' ').append(command.usage.replace("\n", "\\n"))
+//
+//        val prefix = " ${command.parent.name} "
+//        builder.insert(Constants.firstPrefix.length, prefix)
+//        builder.append(" - ").append(command.description)
+//
+//        return builder.toString()
+//    }
```
Comment (DRSchlaubi) [inline src/main/kotlin/dev/schlaubi/votebot/util/Embeds.kt]: remove this as help command isn't needed for slash commansd

File: src/main/kotlin/dev/schlaubi/votebot/util/Embeds.kt [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+    suspend fun KordEvent.respondEmbed(
+        base: EmbedBuilder,
+        creator: suspend EmbedBuilder.() -> Unit = {}
+    ): Message {
+        return respond {
+            creator(base)
+            embed = base
+        }
+    }*/
+}
```
Comment (DRSchlaubi) [inline src/main/kotlin/dev/schlaubi/votebot/util/Embeds.kt]: this can be removed too

**Review by DRSchlaubi — CHANGES_REQUESTED** [review]

File: src/main/kotlin/dev/schlaubi/votebot/commands/ClaimPermissionsCommand.kt (hunk 1 above)
Comment (Niggelgame) [reply src/main/kotlin/dev/schlaubi/votebot/commands/ClaimPermissionsCommand.kt]: yeah, forgot to remove it again since at the beginning it was not null by default

File: src/main/kotlin/dev/schlaubi/votebot/util/Colors.kt (hunk 3 above)
Comment (Niggelgame) [reply src/main/kotlin/dev/schlaubi/votebot/util/Colors.kt]: Should I remove all the non-existing colors?

**Review by DRSchlaubi — APPROVED** [review]

File: src/main/kotlin/dev/schlaubi/votebot/util/Colors.kt (hunk 3 above)
Comment (Niggelgame) [reply src/main/kotlin/dev/schlaubi/votebot/util/Colors.kt]: Or e.g. replace the current LIGHT_GREEN with the branding GREEN

Comment (Niggelgame) [reply src/main/kotlin/dev/schlaubi/votebot/util/Colors.kt]: @DRSchlaubi

**Comment by DRSchlaubi** [comment]
> The base branch requires all commits to be signed. Learn more about signing commits.

DUDE

**Comment by Niggelgame** [comment]
freak i used the wrong git profile

**Comment by DRSchlaubi** [comment]
> freak i used the wrong git profile

just rebase
squash
and sign that

**Comment by Niggelgame** [comment]
actually forgot the squash

but fr took long enough for me to rebase on my own branch... until i found the IJ command to rebase on the root of the branch
